include/sddf/block/blk_shared_queue.h

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2023, UNSW
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+#include "fence.h"
+
+/* Number of buffers the request queue is configured to have. */
+#define BLK_REQ_QUEUE_SIZE 1024
```

Comment:
where does the number come from?

Reply:
typo, "buffers" should change to "slots" here, indicates the number of slots in the request queue. Should be configurable to maximise performance